README.md

```diff
@@ -372,37 +389,37 @@ For updating open swaps, I was successfully able to update 5 positions and recre
+## Frontend Agnosticism
+By using the beacon tokens, it is trivial for any frontend to integrate with Cardano-Swaps. For example, the new Lace wallet by IOHK can integrate the DEX by simply adding support for querying the beacon tokens. They can also add their own user friendly way to create and use swaps. The only requirement is that all frontends/users agree to use the same beacon token standard. There is no need for risky extensions or dedicated frontends in order to use this DEX.
 
 ---
 ## Conclusion
-This DEX protocol has all of the desired properties of a DEX:
+The cardano-swaps protocol has all of the desired properties of a highly scalable DEX:
 
-1. Users maintain delegation control at all times.
-2. Composable atomic swaps allow creating arbitrarily complex swap transactions.
-3. Naturally concurrent and gets more concurrent the more available swaps there are.
-4. There is no impermanent loss since users can declare their desired minimum price.
-5. No secondary token is needed to interact with the DEX; only ADA is needed for the fees and deposits.
-6. Upgrades can happen democratically.
-7. Any frontend can easily add Cardano-Swap support.
 
-Thanks to using beacon tokens, decentralization is no longer limited by the design of DEXs. Instead, the limiting factor is now the off-chain querying. However, innovations in this space are still in the early days. The Koios api is an example of a more decentralized off-chain platform. As the technology improves, the decentralization of this DEX will only improve.
+Thanks to the use of beacon tokens, decentralization is no longer limited by the design of DEXs. Instead, the limiting factor is now the off-chain querying. However, innovations in this space are still in the early days. The Koios api is an example of a more decentralized off-chain platform. As the technology improves, the decentralization of this DEX will only improve.
 
 ---
 ## FAQ
 
-### If all users have direct access to the spending script for that trading pair, why are reference scripts required?
+#### If all users have direct access to the spending script for that trading pair, why are reference scripts used?
+
+To disincentivize beacon bloat. 
+
+The reason is to incentivize users to actually use the DEX as intended. If there were no deposits, users can create a live address with a beacon token and then just leave it open even if they aren't actually using the address anymore (like if there are no swappable assets in it). This "zombie" address will still appear when querying the beacon tokens even though other users can't actually do anything with it. By requiring the deposits, users are incentivized to close unused addresses in order to get the deposits back.
```

Comment:
I could technically use the beacon without reference scripts and just force that the beacon is minted to a utxo containing at least 20 ADA. So this doesn't really answer "why reference scripts.." I forgot to explain it in the original. The idea was:
1) Force a deposit - you can't add a reference script without a minimum amount of ada.
2) Have reference scripts available for use - using reference scripts is usually cheaper than using a local script (although it generally requires more tx memory)

Number 2 may not be necessary since `cardano-reference-scripts` allows sharing reference scripts.

Reply:
how is forcing a deposit enforced? is it possible to alter the code, not be forced to deposit, and still participate in the protocol?

Reply:
The beacon policy can see the entire tx. It looks through the outputs to find what utxo it is going to be stored in. With this information it checks:
1) The utxo is in the proper address.
2) The utxo has the proper reference script for the Dapp.

As a consequence of number 2, since the reference script takes up space, the ledger forces it to be stored with a minimum amount of ADA. This is the deposit. I could change this so that, instead of requiring the utxo to have the proper reference script, the utxo must have a minimum of 20 ADA. The end result is the same.

> is it possible to alter the code, not be forced to deposit, and still participate in the protocol?
Yes, if it doesn't require number 2 and doesn't require the utxo to have a minimum value, then there is no required deposit.

Reply:
ok, then wouldn't it be trivial for an attacker to create a large number of DDOS/zombie addresses, disrupting off-chain querying? 

If so, is there a way to modify/standardize the beacon policy such there is such a thing as "valid" beacons and that they can only be minted to addresses with a certain DDOS-preventative deposit?

Reply:
> wouldn't it be trivial for an attacker to create a large number of DDOS/zombie addresses, disrupting off-chain querying?

Only if I removed the deposit check. Currently, it is impossible to mint a beacon to a dApp address that does not also deposit about 25 ADA (from the size of the reference script). That means if someone were to create 1,000 zombie addresses, they would need to lock up 25,000 ADA for the lifespan of the attack. The dApp script ensures that the deposit can only be reclaimed if the beacon is burned which would stop the attack.

> is there a way to modify/standardize the beacon policy so that there is such a thing as "valid" beacons and that they can only be minted to addresses with an agreed-upon DDOS-preventative deposit?

If the beacon exists at all, then the deposit was guaranteed to be made and the beacon is guaranteed to be protected by the proper spending script.

For generalizing the beacons, we should probably mention in the beacon CIP the need for a deposit to mitigate this attack.

Reply:
I'm saying what if someone altered the code, removed the deposit check, and minted beacons without the deposit?

Are you saying that such beacons would be different from (and always incompatible with) beacons minted by those using scripts *with* deposits?

Reply:
> what if someone altered the code, removed the deposit check, and minted beacons without the deposit?

Different code -> different minting policy -> different beacon policy ID -> entirely different beacon

> Are you saying that such beacons would be different from (and always incompatible with) beacons minted by those using scripts with deposits?

Yes. The beacon policy ID is the hash of the minting policy. If the code is different, then the minting policy ID is guaranteed to be different. So even if they use the same token name, the policy ID would be different and therefore it would be an entirely different token.